### PR TITLE
Fix Reset Filters

### PR DESCRIPTION
**Pull request by Rasbin**
This PR should fix : `Reset filters` button should clear filters after it is clicked without needing to click the `Search` button after the `Reset filter` is clicked.

### CODE REVIEW

File: app/pages/search/controls/SearchControlsContainer.js [hunk 1]
```diff
@@ -62,7 +62,7 @@ class UnconnectedSearchControlsContainer extends Component {
     const { filters, position } = this.props;
     let hasFilters = Boolean(position);
     ['freeOfCharge', 'end', 'distance', 'duration', 'purpose', 'start', 'unit'].forEach((key) => {
-      if (filters[key]) {
+      if (filters[key] && filters[key] !== '0') {
```
Comment (rBoost) [inline app/pages/search/controls/SearchControlsContainer.js]: Why is this needed?

**Comment by rBoost** [comment]
> I think you misunderstand the ticket, the implement should be like dispatch and resolve the reset action immediately after you press the button, not after you hit Search.
> Therefore, this fix is not valid, either make no sense to add the extra condition that look strange

If you check the `handleSearch` method you see that this PR does exactly that. My only question is what does the change to `hasAdvancedFilters` do, why is the change needed?

File: app/pages/search/controls/SearchControlsContainer.js (hunk 1 above)
Comment (Rasbin) [reply app/pages/search/controls/SearchControlsContainer.js]: To remove the display of `Reset Filters` button after it is clicked I had to add that line. It seems like the filter value of duration has been provided zero (in number, not in string) and treated at false and affected. So, to fix that I added that code. 
screen shot 2019-03-06 at 10 04 30

Comment (rBoost) [reply app/pages/search/controls/SearchControlsContainer.js]: ok. I think we need to improve how search filters are put to the url. I don't think empty filters should be included in the url at all. But that's an entirely different task.

**Comment by midori0507** [comment]
> > I think you misunderstand the ticket, the implement should be like dispatch and resolve the reset action immediately after you press the button, not after you hit Search.
> > Therefore, this fix is not valid, either make no sense to add the extra condition that look strange
> 
> If you check the `handleSearch` method you see that this PR does exactly that. My only question is what does the change to `hasAdvancedFilters` do, why is the change needed?

The thing is 

> > I think you misunderstand the ticket, the implement should be like dispatch and resolve the reset action immediately after you press the button, not after you hit Search.
> > Therefore, this fix is not valid, either make no sense to add the extra condition that look strange
> 
> If you check the `handleSearch` method you see that this PR does exactly that. My only question is what does the change to `hasAdvancedFilters` do, why is the change needed?

Yes you're right, i didn't notice small line change below.
Anyway, what @Rasbin trying to do here (small thing left) is the reset filter button suppose to disappear after clicked but it's not (randomly)
I suggest writing unit tests for that behavior, if the unit test passing but the issue still persist, we got problem else where, not in the code of this function

File: app/pages/search/controls/SearchControlsContainer.js (hunk 1 above)
Comment (midori0507) [reply app/pages/search/controls/SearchControlsContainer.js]: We discuss and agree that this condition is not making any change (unnecessary)

File: app/pages/search/controls/SearchControlsContainer.js [hunk 2]
```diff
@@ -121,6 +121,7 @@ class UnconnectedSearchControlsContainer extends Component {
     }
 
     this.handleFiltersChange(emptyFilters);
+    this.handleSearch(emptyFilters);
```
Comment (midori0507) [inline app/pages/search/controls/SearchControlsContainer.js]: Can you write unit test for this, something as 
`when reset button was click, trigger handleSearch`

File: app/pages/search/controls/SearchControlsContainer.js (hunk 1 above)
Comment (Rasbin) [reply app/pages/search/controls/SearchControlsContainer.js]: Now, it is also working without the extra condition. It was not working before. Will investigate a bit and commit after checking if it works correct consistently.

File: app/pages/search/controls/SearchControlsContainer.js (hunk 2 above)
Comment (Rasbin) [reply app/pages/search/controls/SearchControlsContainer.js]: It already failed with couple of manual testing. I will find the cause and fix the bug first.

**Comment by rBoost** [comment]
@Rasbin It seems that there are some conflicts that need to be resolved before merging this issue. 
Also, we should make sure that test coverage doesn't go down.

File: app/pages/search/controls/SelectControl.js [hunk 3]
```diff
@@ -13,7 +13,6 @@ class SelectControl extends React.Component {
     if (isArray(value)) {
       return value.map(item => options.find(option => option.value === item));
     }
-
```
Comment (midori0507) [inline app/pages/search/controls/SelectControl.js]: Can you remove this ?

File: app/pages/search/controls/SearchControlsContainer.js (hunk 2 above)
Comment (midori0507) [reply app/pages/search/controls/SearchControlsContainer.js]: But still, write unit tests for it

Comment (Rasbin) [reply app/pages/search/controls/SearchControlsContainer.js]: I meant the reset button was not working as intended with couple of manual checks and wanted to fix it first. Missed to write it clearly before. Sorry, My bad!

File: app/pages/search/controls/SelectControl.js (hunk 3 above)
Comment (Rasbin) [reply app/pages/search/controls/SelectControl.js]: Done!

**Comment by Rasbin** [comment]
@rBoost Can you check it now if it is okay?

**Comment by rBoost** [comment]
Looks good 👍
